### ISSUE TITLE
[git propose] Allow passing a `--title` and `--body`

### ISSUE THREAD

**Issue by ruudk** [git-town/git-town#3207]
I'd like to be able to pass a title and body when proposing a change (creating a pull request).

GitHub supports this, by sending the following query parameters: `?expand=1&title=title&body=body`.

I'd like to use this in my custom script that automatically finds a title (using AI) and a body (by taking all commit messages).

It would be great if the body can also be passed via stdin like `gh` cli does:

```
echo "long\nbody" | gh pr create --web --title "$TITLE" --body-file -
```

**Comment by ruudk** [comment]
When you create a PR on GitHub and there is only 1 commit, GitHub will use the commit title as the title of the PR.

Maybe Git Town can do the same. If there is only 1 commit, then use that as a title. If there are more than 1 commit, ask for a custom title.

**Comment by ruudk** [comment]
🥰🚀🤩🥳👍💪

**Comment by kevgo** [comment]
This is now shipped in [v14.3.0](https://github.com/git-town/git-town/releases/tag/v14.3.0). Enjoy!

**Comment by ruudk** [comment]
Just wanted to let you know that this feature works perfectly! 

Together with [git town config get-parent](https://github.com/git-town/git-town/issues/3628).

I created this `git-create-pr` script:

```bash
#/!/bin/bash

set -e

brew list gh &>/dev/null || brew install gh
brew list fzf &>/dev/null || brew install fzf

BRANCH=$(git branch --show-current)

if [ "$BRANCH" == "main" ]; then
  echo "You are on the main branch. You cannot run this here."
  exit 1
fi

BRANCHPOINT=$(git town config get-parent | tr -d '\n')
COMMITS=$(git log --oneline --graph $BRANCHPOINT.. | wc -l | tr -d ' ')

if [ "$COMMITS" -eq 0 ]; then
  echo "This branch is empty."
  exit 0
fi

FIXUPS=$(echo "$COMMITS" | grep "fixup!\|squash!" | wc -l)
if [ $FIXUPS -gt 0 ]; then
  echo "The branch contains fixup commits."
  exit 1
fi

CUSTOM_TITLE="Enter custom title"

if [ "$COMMITS" -eq 1 ]; then
  TITLE=$(git log -1 --reverse --no-merges --pretty=format:%s $BRANCHPOINT..)
  DESCRIPTION=$(git log --reverse --no-merges --pretty='%b' $BRANCHPOINT..)
else
  TITLE=$({ git log --reverse --no-merges --pretty=format:%s $BRANCHPOINT..; echo ""; echo "$CUSTOM_TITLE"; } | fzf --ansi --no-sort --header-first --header="Please choose the title of the PR")
  if [ "$TITLE" == "$CUSTOM_TITLE" ]; then
    read -p "Enter title: " TITLE
  fi
  DESCRIPTION=$(git log --reverse --no-merges --pretty='### %B' $BRANCHPOINT..)
fi

echo "$DESCRIPTION" | git town propose --title "$TITLE" --body-file -
```

🚀
